.github/scripts/setup-workflow-monitor.py

```diff
@@ -4,28 +4,48 @@
 import sys
 
 from common import manager_ci_dir, set_fabric_firesim_pem
-from ci_variables import ci_workflow_run_id, ci_personal_api_token
+from platform_lib import Platform, get_platform_enum
+from ci_variables import ci_azure_sub_id, ci_azure_client_id, ci_azure_tenant_id, ci_azure_client_secret, \
+    ci_workflow_run_id, ci_personal_api_token
 
-def setup_workflow_monitor(max_runtime: int) -> None:
+def generate_azure_credited_env():
+    run(f"echo 'export AZURE_CREDITED_ENV=true' >> ~/azure_env.sh")
+    run(f"echo 'export AZURE_SUBSCRIPTION_ID={ci_azure_sub_id}' >> ~/azure_env.sh")
+    run(f"echo 'export AZURE_CLIENT_ID={ci_azure_client_id}' >> ~/azure_env.sh")
+    run(f"echo 'export AZURE_TENANT_ID={ci_azure_tenant_id}' >> ~/azure_env.sh")
+    run(f"echo 'export AZURE_CLIENT_SECRET={ci_azure_client_secret}' >> ~/azure_env.sh")
+
+def setup_workflow_monitor(platform: Platform, max_runtime: int) -> None:
     """ Performs the prerequisite tasks for all CI jobs that will run on the manager instance
 
     max_runtime (hours): The maximum uptime this manager and its associated
         instances should have before it is stopped. This serves as a redundant check
         in case the workflow-monitor is brought down for some reason.
+    
+    platform: Enum that indicates either 'aws' or 'azure' currently. Describes the current platform 
+        from which CI is being run from.
     """
     with cd(manager_ci_dir):
+        # This generates a file that can be sourced to get all the right keys / ids to run any of the
+        # Azure jobs. This is always generated and sourced, but is completely unused by the AWS platform.
+        generate_azure_credited_env()
```

Comment:
Can you just use a `with prefix(...)` call around the `run` to setup the right environment? Wouldn't that then get passed to the subshell that runs the screen (removing the need for a file)?

Reply:
When I tested this in CI, running exports / running with a prefix did not result in the screen job inheriting the environment variables, which is unfortunately the only (feasible) way for us internally to pass credentials to CI.